### DISCUSSION TITLE
Not able to display custom linked work item

### DISCUSSION THREAD

**Discussion by JaydeepUniverse** [rfennell/AzurePipelines#1138] [General]
Generally in the work item we can see 2 or 3 types of linked work items: Parent, Child, Related
and in the related work item we have diff. kind of work item types like  - bug, task, test plan, test case

We have created custom linked work item type called: Tested By for test plan and test case Ex,

![image](https://user-images.githubusercontent.com/25937808/141428445-15885516-1743-452b-9bb8-0f1cd0d7edde.png)

To get the parent work item details we're using below code
```
{{#forEach this.relations}}
    {{#if (contains this.attributes.name 'Parent')}} 
        {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}} 
            #{{this.id}} 
        {{/with}} 
    {{/if}} 
{{/forEach}}
```

and similarly I created another code for Tested By as below, But then issue is those tested by linked work items are not showing up
```
{{#forEach this.relations}}
    {{#if (contains this.attributes.name 'Tested By')}} 
        {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}} 
            #{{this.id}} 
        {{/with}} 
    {{/if}} 
{{/forEach}}
```

We have few work items for which "Tested By" linked work item is not implemented yet, so to cross check other behavior I have noticed is that, I have created new code to get "Related" work items, what I get is related work items but other work items which are type of "Test case" or "Test plan" are not coming in the list
```
{{#forEach this.relations}}
    {{#if (contains this.attributes.name 'Related')}} 
        {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}} 
            #{{this.id}} 
        {{/with}} 
    {{/if}} 
{{/forEach}}
```

Kindly advise how we can resolve this?

Let me know if any other details are needed

Thanks,
Jaydeep

**Comment by rfennell** [comment]
A few questions....

- Can I confirm you are adding links to tests using the built in link types of 'tests' and 'tested by', as I do in this example
![image](https://user-images.githubusercontent.com/324718/141441254-f83d77bd-0dfc-4198-a57a-2e37df9152e9.png)

- The reason I ask is if you are using standard relations I can probably make a change, if it is custom relation (specific to you) I can't.

**Comment by JaydeepUniverse** [comment]
@rfennell Apologies, I provided wrong information.

Yes, "Tested By" is existing one we are using, we have not configured any field custom.

**Reply by rfennell** [reply]
Ok, I can extend the logic to add 'tested' items, I will probably add them to a new array as they are not strictly 'related'. I will create an issue to track the work.

**Reply by JaydeepUniverse** [reply]
Ok, Thanks for this help @rfennell
